### PR TITLE
Improve docstring of AsTable

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2850

### CODE REVIEW

File: src/other/utils.jl [hunk 1]
```diff
@@ -1,8 +1,54 @@
...
+julia> select(df2, :a_b_identity => AsTable)
+3×2 DataFrame
+ Row │ a      b     
+     │ Int64  Int64 
+─────┼──────────────
+   1 │     1     11
+   2 │     2     12
+   3 │     3     13
+
+julia> select(df, AsTable([:a, :b]) => ByRow(nt -> map(x -> x^2, nt)) => AsTable)
```
Comment (bkamins) [inline src/other/utils.jl:43]:
```suggestion
julia> select(df1, AsTable([:a, :b]) => ByRow(nt -> map(x -> x^2, nt)) => AsTable)
```

File: src/other/utils.jl [hunk 2]
```diff
@@ -1,8 +1,54 @@
...
-wrapped selector should be passed as a `NamedTuple` to the function.
+A type having a special meaning in `source => transformation => destination`
+selection operations supported by [`combine`](@ref), [`select`](@ref), [`select!`](@ref),
+[`transform`](@ref), [`transform!`](@ref), [`subset`](@ref), and [`subset!`](@ref).
+
+If `AsTable(cols)` is used in `source` position it signals that the columns selected
+by the wrapped selector `cols` should be passed as a `NamedTuple` to the function.
+
+If `AsTable` is used in `destination` position it means that the result of the operation
+is a vector of containers that be expanded into multiple columns using `keys` to get column names.
```
Comment (kleinschmidt) [inline src/other/utils.jl:12]:
```suggestion
If `AsTable` is used in `destination` position it means that the result of the operation
is a vector of containers (or a single container if `ByRow` is used) that should be expanded 
into multiple columns using `keys` to get column names.
```

**Review by kleinschmidt — APPROVED** [review]
Generally looks great, thanks for the quick fix!!  One tiny suggestion

**Review by nalimilan — APPROVED** [review]

File: src/other/utils.jl [hunk 3]
```diff
@@ -1,8 +1,55 @@
...
+A type having a special meaning in `source => transformation => destination`
+selection operations supported by [`combine`](@ref), [`select`](@ref), [`select!`](@ref),
+[`transform`](@ref), [`transform!`](@ref), [`subset`](@ref), and [`subset!`](@ref).
+
+If `AsTable(cols)` is used in `source` position it signals that the columns selected
+by the wrapped selector `cols` should be passed as a `NamedTuple` to the function.
+
+If `AsTable` is used in `destination` position it means that the result of the `transformation` operation
+is a vector of containers (or a single container if `ByRow` is used) that should be expanded 
+into multiple columns using `keys` to get column names.
```
Comment (bkamins) [inline src/other/utils.jl:13]:
```suggestion
If `AsTable` is used in `destination` position it means that the result of
the `transformation` operation is a vector of containers
(or a single container if `ByRow(transformation)` is used)
that should be expanded  into multiple columns using `keys` to get column names.
```

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**